client-lib/java/src/main/java/com/abcxyz/jvs/JvsClient.java

```diff
@@ -34,52 +36,80 @@
 @RequiredArgsConstructor(access = AccessLevel.PACKAGE)
 @Slf4j
 public class JvsClient {
-  // This postfix is added by the cli tool when creating breakglass tokens.
-  private static final String UNSIGNED_POSTFIX = ".NOT_SIGNED";
+  // This is the HMAC key to use for creating breakglass tokens. Breakglass
+  // tokens are already "unverified", so having this static secret does not
+  // introduce additional risk, and breakglass is disabled by default.
+  protected static final String BREAKGLASS_HMAC_SECRET =
+      "BHzwNUbxcgpNoDfzwzt4Dr2nVXByUCWl1m8Eq2Jh26CGqu8IQ0VdiyjxnCtNahh9";
 
-  // this category is added by the cli tool when creating breakglass tokens.
-  private static final String BREAKGLASS_CATEGORY = "breakglass";
+  // This is the justification category set for breakglass tokens.
+  private static final String BREAKGLASS_JUSTIFICATION_CATEGORY = "breakglass";
 
   private final JwkProvider provider;
   private final boolean allowBreakglass;
 
-  public DecodedJWT validateJWT(String jwtString) throws JwkException {
-    DecodedJWT jwt = JWT.decode(jwtString);
+  /**
+   * This parses the given token as a breakglass token. If the token is not a JWT, not HS256, or
+   * missing breakglass justifications, it returns null. If the token fails to verify, it throws an
+   * exception.
+   */
+  private DecodedJWT parseBreakglassToken(String tokenStr) throws JwkException {
+    DecodedJWT token = JWT.decode(tokenStr);
 
-    // Handle unsigned tokens.
-    if (jwtString.endsWith(UNSIGNED_POSTFIX)) {
-      if (unsignedTokenValidAndAllowed(jwt)) {
-        return jwt;
-      } else {
-        throw new JwkException("Token unsigned and could not be validated.");
+    if (!Header.JWT_TYPE.equals(token.getType())
+        || !SignatureAlgorithm.HS256.getValue().equals(token.getAlgorithm())) {
+      log.debug("token is not breakglass");
+      return null;
+    }
+
+    try {
+      Algorithm alg = Algorithm.HMAC256(BREAKGLASS_HMAC_SECRET);
+      alg.verify(token);
+    } catch (SignatureVerificationException e) {
+      log.error("failed to verify breakglass token: {}", e);
+      throw new JwkException("failed to parse breakglass jwt", e);
+    }
+
+    List<Justification> justifications = token.getClaim("justs").asList(Justification.class);
+    for (Justification justification : justifications) {
+      if (BREAKGLASS_JUSTIFICATION_CATEGORY.equals(justification.getCategory())) {
+        return token;
       }
     }
 
+    log.error("token is breakglass, but is missing breakglass justification");
```

Comment:
In Go, we will return error if no breakglass explanation is found. So should we throw an exception here?

Reply:
Updated